components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java

```diff
@@ -154,7 +150,11 @@ public boolean validateScope(OAuthTokenReqMessageContext tokReqMsgCtx)
                 return false;
             }
             List<String> grantedScopeList = Stream
-                    .concat(Arrays.stream(grantedScopes), Arrays.stream(grantedInternalScopes))
+                    .concat(grantedScopes == null
+                            ? Stream.empty()
+                            : Arrays.stream(grantedScopes), grantedInternalScopes == null
+                            ? Stream.empty()
+                            : Arrays.stream(grantedInternalScopes))
```

Comment:
What if we do this just before creating the list?
It would be much readable IMO